webapp/components/interop.js

```diff
@@ -200,6 +201,28 @@ class InteropDataManager {
     }
   }
 
+  #addInvestigationScore(summaryScore, date) {
+    if (this.investigationScores) {
+      let totalInvestigationScore = 0;
+      this.investigationScores.forEach(info => {
+        let start = info.scores_over_time.length - 1;
+        let areaScore = 0;
+        for (let i = start; i >= 0; i--) {
+          if (date < new Date(info.scores_over_time[i].date)) {
+            continue;
+          }
+          areaScore = info.scores_over_time[i].score;
+          break;
+        }
```

Comment:
I think this could be simpler/clearer using `findLast()`. Untested suggestion:

```suggestion
        let entry = info.scores_over_time.findLast((entry) => {
          return date >= new Date(entry.date);
        });
        if (entry) {
          totalInvestigationScore += entry.score;
        }
```

Reply:
Much more readable - I implemented it 👍